pkg/loop/reportingplugins/grpc.go

```diff
@@ -0,0 +1,81 @@
+package reportingplugins
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-plugin"
+	"google.golang.org/grpc"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/loop"
+	"github.com/smartcontractkit/chainlink-relay/pkg/loop/internal"
+	"github.com/smartcontractkit/chainlink-relay/pkg/types"
+)
+
+// PluginServiceName is the name for [types.PluginClient]/[NewGRPCService].
+const PluginServiceName = "plugin-service"
+
+func ReportingPluginHandshakeConfig() plugin.HandshakeConfig {
+	return plugin.HandshakeConfig{
+		MagicCookieKey:   "CL_PLUGIN_GENERIC_MAGIC_COOKIE",
+		MagicCookieValue: "2ad981747cd86c4ab3e23170970020fd",
+	}
+}
+
+type ProviderServer[T types.PluginProvider] interface {
+	types.ReportingPluginServer[T]
+	ConnToProvider(conn grpc.ClientConnInterface, broker internal.Broker, brokerConfig loop.BrokerConfig) T
+}
+
+// GRPCService is the loopp interface for a plugin that can
+// run an arbitrary product in the core node. By specifying
+// `T`, server instances can request a specific provider type.
+type GRPCService[T types.PluginProvider] struct {
+	plugin.NetRPCUnsupportedPlugin
+
+	loop.BrokerConfig
+
+	PluginServer ProviderServer[T]
+
+	pluginClient *internal.ReportingPluginServiceClient
+}
+
+type serverAdapter func(context.Context, types.ReportingPluginServiceConfig, grpc.ClientConnInterface, types.ErrorLog) (types.ReportingPluginFactory, error)
+
+func (g serverAdapter) NewReportingPluginFactory(ctx context.Context, config types.ReportingPluginServiceConfig, conn grpc.ClientConnInterface, errorLog types.ErrorLog) (types.ReportingPluginFactory, error) {
```

Comment:
Why `g` for `serverAdapter`, rather than `s` or `a`?

Reply:
Ugh, this is a holdover from when this was called `grpcServerAdapter`